plugins/inputs/tacacs/tacacs.go

```diff
@@ -0,0 +1,147 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package tacacs
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/nwaples/tacplus"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Tacacs struct {
+	Servers         []string        `toml:"servers"`
+	Username        config.Secret   `toml:"username"`
+	Password        config.Secret   `toml:"password"`
+	Secret          config.Secret   `toml:"secret"`
+	RemAddr         string          `toml:"request_ip"`
+	ResponseTimeout config.Duration `toml:"response_timeout"`
+	Log             telegraf.Logger `toml:"-"`
+	client          tacplus.Client
+}
+
+//go:embed sample.conf
+var sampleConfig string
+
+func (t *Tacacs) SampleConfig() string {
+	return sampleConfig
+}
+
+func (t *Tacacs) Init() error {
+	if len(t.Servers) == 0 {
+		t.Servers = []string{"127.0.0.1:49"}
+	}
+
+	t.client = tacplus.Client{}
+
+	return nil
+}
+
+func (t *Tacacs) Gather(acc telegraf.Accumulator) error {
+	var wg sync.WaitGroup
+
+	for _, server := range t.Servers {
+		wg.Add(1)
+		go func(server string) {
+			defer wg.Done()
+			acc.AddError(t.pollServer(acc, server))
+		}(server)
+	}
+
+	wg.Wait()
+	return nil
+}
+
+func (t *Tacacs) pollServer(acc telegraf.Accumulator, server string) error {
+	// Create the fields for this metric
+	tags := map[string]string{"source": server}
+	fields := make(map[string]interface{})
+
+	secret, err := t.Secret.Get()
+	if err != nil {
+		return fmt.Errorf("getting secret failed: %w", err)
+	}
+	defer config.ReleaseSecret(secret)
+
+	username, err := t.Username.Get()
+	if err != nil {
+		return fmt.Errorf("getting username failed: %w", err)
+	}
+	defer config.ReleaseSecret(username)
+
+	password, err := t.Password.Get()
+	if err != nil {
+		return fmt.Errorf("getting password failed: %w", err)
+	}
+	defer config.ReleaseSecret(password)
+
+	t.client.Addr = server
+	t.client.ConnConfig = tacplus.ConnConfig{
+		Secret: secret,
+	}
+
+	// Initialize the AuthStart data
+	testAuthStart := &tacplus.AuthenStart{
+		Action:        tacplus.AuthenActionLogin,
+		AuthenType:    tacplus.AuthenTypeASCII,
+		AuthenService: tacplus.AuthenServiceLogin,
+		PrivLvl:       1,
+		Port:          "heartbeat",
+		RemAddr:       t.RemAddr,
+	}
```

Comment:
These can still be moved to Init, you can make an array of clients. Then the `pollServer` would get the `client tacplus.Client` or just the id..